### ISSUE TITLE
HM: Put coming and going edge connectors on different positions

### ISSUE THREAD

**Issue by urapadmin** [arch-kiosk/arch-kiosk-office#2521]
Meanwhile the colour version reveals what is going on in the line crowding:
![image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/660180ba-af87-4faf-94d2-f08dd805a7d4)
![image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/06f7493d-6756-431f-a7d9-10a0daee6b94)

So that's not a mistake of the algorithm, thank god, but an unfortunate outcome. I am wondering if I could allow the line connectors to spread a bit more and have the coming and the going connectors on different positions.  I think I'll make that a ticket.

_Originally posted by @urapadmin in https://github.com/arch-kiosk/arch-kiosk-office/issues/2513#issuecomment-1924200510_

**Comment by urapadmin** [comment]
### Plugin version 0.5
not as trivial as I thought. But I think I achieved it:


![Image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/c25435bb-c012-4093-ac5b-203a46582a86)


I also moved incoming and outgoing connectors closer together if there is no middle connector:


![Image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/ac5a4a45-b0f5-459c-ade5-0ee29cf8d7e5)

And 


![Image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/de584c23-a91f-48a2-aaa6-7ca22eb57aa6)


And I made sure that single connectors (only one incoming or outgoing) are always almost in the middle of the node:


![Image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/c51a990b-88d4-44fb-ac4e-467dfd92bf50)

**Comment by luizaogs** [comment]
Looks good to me.

**Comment by urapadmin** [comment]
Not in PASU
![image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/939d3fee-9d8a-49e6-abc2-e16ed6ad391f)
Here it does not seem to work at all

**Comment by urapadmin** [comment]
also: Why is the font here soooo bad? But that is a different issue:

**Comment by luizaogs** [comment]
Well that's weird. I checked both MKAP and URAP and they seemed fine. Also PVD, I think...

**Comment by urapadmin** [comment]
ja, it looks exactly like the same constellation of nodes and edges that we had in MKAP and there it is solved. Hm. Neverending edge Wirrwarr.

**Comment by urapadmin** [comment]
same problem in URAP:
![image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/763c6103-58c6-476b-be95-5458b866f9fb)

happens when opening FA-058

**Comment by luizaogs** [comment]
Hmpf so it really seems like it differs by HM sometimes, no?

**Comment by urapadmin** [comment]
### plugin v0.8.0
- this is a special constellation where there is no incoming middle edge straight from the node above, so I try to use the middle position for the edges coming from left and/or right. And if that coincides with a node above that has an outgoing edge that likewise is trying to use the vacant middle position, they fell on the same horizontal position. Sounds pretty rare, but happens often enough. URAP FA-058 should look like this now:
![image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/9f749b36-26bf-4a4d-a350-706696b2fc4a)

And all the other examples should be better now, too. I hope I have not created a new issue with a new constellation somewhere else now. But it is possible, so keep your eyes open for such things in all projects.

**Comment by urapadmin** [comment]
I found something myself:
![image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/8098ca44-ffe3-4269-8a69-ff23e94c162d)
